src/tests/functional/plugin/conformance/test_runner/conformance_infra/src/main.cpp

```diff
@@ -4,24 +4,52 @@
 
 #include <signal.h>
 #ifdef _WIN32
-#include <process.h>
+#    include <process.h>
 #endif
 
-#include "gtest/gtest.h"
-
 #include "common_test_utils/file_utils.hpp"
+#include "conformance.hpp"
 #include "functional_test_utils/skip_tests_config.hpp"
 #include "functional_test_utils/summary/environment.hpp"
-
-#include "read_ir_test/read_ir.hpp"
 #include "gflag_config.hpp"
-#include "conformance.hpp"
+#include "gtest/gtest.h"
+#include "read_ir_test/read_ir.hpp"
 
+#ifdef ENABLE_CONFORMANCE_PGQL
+#    include "common_test_utils/postgres_link.hpp"
+
+void RegisterTestCustomQueries(void) {
+    std::map<std::string, std::string>& extTestQueries = *::PostgreSQLLink::get_ext_test_queries();
+    std::map<std::string, std::string>& extTestNames = *::PostgreSQLLink::get_ext_test_names();
+
+    std::string testName("checkPluginImplementation");
+    extTestQueries[testName + "_ON_START"] = "OpImplCheck_CheckPluginImpl($__test_id, '$opName', '$opSet', "
+                                             "'$targetDevice', '$targetDeviceArch', '$config', $__is_temp)";
+    extTestQueries[testName + "_ON_END"] = "OpImplCheck_CheckPluginImpl($__test_ext_id, $__test_id)";
+    extTestQueries[testName + "_ON_REFUSE"] =
+        "OpImplCheck_CheckPluginImpl($__test_id)";  // Query expected in case of a refused results
+    extTestNames[testName] = "$opName";
+
+    testName = "ReadIR";
```

Comment:
Just question: what if we will extend conformance suite by other ones tests?

Reply:
It isn't a problem, you may add them, combine, etc... This way is used for storing an additional information for each test case.

In case you don't provide extended queries you may render a report where you will see just simple hierarchy: Test Suite -> Test Case = Test Result